### PR TITLE
Updates based on changes to podresource API and CRD spec

### PR DESCRIPTION
**Pull request by swatisehgal**
- KEP: https://github.com/kubernetes/enhancements/pull/1884/files
 - Obtaining cpu and device information from kubelet podresource API.
   With the introduction of GetAvailableResources() endpoint in podresource
   API, we can now account for all the devices known to kubelet in a more
   generic manner and no longer need to rely on SR-IOV config.
 - Remove CRI support from this repository due to all its drawbacks
   compared to podresource API
 - Updates to create zone based NodeResourceTopology CRD object.
   More details here: https://docs.google.com/document/d/12kj3fK8boNuPNqob6F_pPU9ZTaNEnPGaXEooW1Cilwg/edit

### CODE REVIEW

File: pkg/exporter/exporter.go [hunk 1]
```diff
@@ -45,17 +45,19 @@ func NewExporter(tmPolicy string) (*CRDExporter, error) {
...
 
-func (e *CRDExporter) CreateOrUpdate(namespace string, resources []v1alpha1.NUMANodeResource) error {
-	log.Printf("Exporter Update called NodeResources is: %+v", resources)
+func (e *CRDExporter) CreateOrUpdate(namespace string, zones v1alpha1.ZoneMap) error {
+	log.Printf("Exporter Update called NodeResources is: %+v", zones)
 
 	nrt, err := e.cli.TopologyV1alpha1().NodeResourceTopologies(namespace).Get(context.TODO(), e.hostname, metav1.GetOptions{})
 	if errors.IsNotFound(err) {
+		policy := []string{}
+		policy = append(policy, e.topologyManagerPolicy)
```
Comment (ffromani) [inline pkg/exporter/exporter.go]: minor... this can be merged

File: pkg/exporter/exporter.go [hunk 2]
```diff
@@ -45,17 +45,19 @@ func NewExporter(tmPolicy string) (*CRDExporter, error) {
...
 		nrtNew := v1alpha1.NodeResourceTopology{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: e.hostname,
 			},
-			Nodes:          resources,
-			TopologyPolicy: e.topologyManagerPolicy,
+			Zones:          zones,
+			TopologyPolicy: policy,
```
Comment (ffromani) [inline pkg/exporter/exporter.go]: or even inlined here, something like

```
TopologyPolicy: []string{
        e.topologyManagerPolicy,
},
```

File: README.md [hunk 3]
```diff
@@ -115,14 +87,41 @@ To test the working of exporter, deploy test deployment that request resources
...
 ```
 
-NOTE
-This exporter assumes the cluster has devices configured using [Sample device plugin](https://github.com/swatisehgal/sample-device-plugin)
-
 ## Limitations
 
-Due to the current (2020, July) limitations of both CRI and podresources APIs for the purposes of the resource-topology-exporter,
-we have explicit support for most commonly used device plugins.
+* RTE assumes the devices are not created dynamically.
```
Comment (ffromani) [inline README.md:92]: for the future: we may want to crosscheck and hint that AFAIK _neither_ kubelet (1.20) handles hotplug, so this limitation is actually dwarfed by kube's limitation.

File: manifests/sample-devices/test-pod-deviceA-2.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod-2
+  namespace: rte
+spec:
+  nodeName: cnfd2-worker-0.fci1.kni.lab.eng.bos.redhat.com
```
Comment (ffromani) [inline manifests/sample-devices/test-pod-deviceA-2.yaml]: better to avoid this

File: manifests/sample-devices/test-pod-deviceA-3.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod-3
+  namespace: rte
+spec:
+  nodeName: cnfd0-worker-0.fci1.kni.lab.eng.bos.redhat.com
```
Comment (ffromani) [inline manifests/sample-devices/test-pod-deviceA-3.yaml]: same

File: manifests/sample-devices/test-pod-deviceA.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod-1
+  namespace: rte
+spec:
+  nodeName: cnfd2-worker-0.fci1.kni.lab.eng.bos.redhat.com
```
Comment (ffromani) [inline manifests/sample-devices/test-pod-deviceA.yaml]: same

**Review by ffromani — APPROVED** [review]
/lgtm
few minor comments inside
Overall looks ok, I think we can rearchitect the flows a bit in the near future

**Comment by swatisehgal** [comment]
@fromanirh Thanks for your review, I have made updates based on the comments. Add good to squash and merge?

**Comment by ffromani** [comment]
yep, go ahead. We can re-evaluate the internals later.

**Pull request merged**